package/MDAnalysis/core/universe.py

```diff
@@ -627,26 +636,42 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
         from ..coordinates.memory import MemoryReader
 
         if not isinstance(self.trajectory, MemoryReader):
-            # Try to extract coordinates using Timeseries object
-            # This is significantly faster, but only implemented for certain
-            # trajectory file formats
-            try:
-                coordinates = self.trajectory.timeseries(
-                    self.atoms, start=start, stop=stop, step=step, order='fac')
-            # if the Timeseries extraction fails,
-            # fall back to a slower approach
-            except AttributeError:
-                n_frames = len(range(
-                    *self.trajectory.check_slice_indices(start, stop, step)
-                ))
-                pm_format = '{step}/{numsteps} frames copied to memory (frame {frame})'
-                pm = ProgressMeter(n_frames, interval=1,
-                                   verbose=verbose, format=pm_format)
-                coordinates = []  # TODO: use pre-allocated array
-                for i, ts in enumerate(self.trajectory[start:stop:step]):
-                    coordinates.append(np.copy(ts.positions))
-                    pm.echo(i, frame=ts.frame)
-                coordinates = np.array(coordinates)
+            n_frames = len(range(
+                *self.trajectory.check_slice_indices(start, stop, step)
+            ))
+            n_atoms = len(self.atoms)
+            pm_format = '{step}/{numsteps} frames copied to memory (frame {frame})'
+            pm = ProgressMeter(n_frames, interval=1,
+                               verbose=verbose, format=pm_format)
+            coordinates = np.zeros((n_frames, n_atoms, 3), dtype=np.float32)
+            # TODO: This assumes constant...
+            ts = self.trajectory.ts
+            has_vels = ts.has_velocities
+            has_fors = ts.has_forces
+            has_dims = ts.dimensions is not None
+
+            if has_vels:
```

Comment:
ternary operator for readability?
```python
velocities = np.zeros_like(coordinates) if has_vels or None
```
Does Python short-circuit, i.e., is the creation of `np.zeros_like(coordinates)` skipped if not needed?

Reply:
Python (at least 3.5.3) properly short-circuits and does not execute the function in the skipped part:
```python
In [1]: def dome():
   ...:     print("did it")
   ...:     return 10
   ...:

In [2]: x = dome() if True else 0
did it

In [3]: x = dome() if False else 0
```